lib/iris/tests/unit/coords/test_AuxCoord.py

```diff
@@ -0,0 +1,613 @@
+        parent_coord = AuxCoord([1., 2., 3.])
+        sub_coord = parent_coord[:]
+        values_before_change = sub_coord.points.copy()
+        parent_coord.points[:] = -999.9
+        self.assertArrayEqual(sub_coord.points, values_before_change)
+
+    def test_dtypes(self):
+        # Index coords with all combinations of real+lazy points+bounds, and
+        # either an int or floating dtype.
+        # Check that dtypes are preserved in all cases, taking dtypes directly
+        # from the core points and bounds arrays (as we have no masking).
+        for (main_coord, points_type_name, bounds_type_name) in \
+                coords_all_dtypes_and_lazynesses(self, AuxCoord):
+
+            sub_coord = main_coord[:2, 1]
+
+            coord_dtype = main_coord.dtype
+            msg = ('Indexing main_coord of dtype {} '
+                   'with {} points and {} bounds '
+                   'changed dtype of {} to {}.')
+
+            sub_points = sub_coord.core_points()
+            self.assertEqual(
+                sub_points.dtype, coord_dtype,
+                msg.format(coord_dtype,
+                           points_type_name, bounds_type_name,
+                           'points', sub_points.dtype))
+
+            if bounds_type_name is not 'no':
+                sub_bounds = sub_coord.core_bounds()
+                self.assertEqual(
+                    sub_bounds.dtype, coord_dtype,
+                    msg.format(coord_dtype,
+                               points_type_name, bounds_type_name,
+                               'bounds', sub_points.dtype))
+
+    def test_lazyness(self):
+        # Index coords with all combinations of real+lazy points+bounds, and
+        # either an int or floating dtype.
+        # Check that laziness is preserved in all cases.
+        for (main_coord, points_type_name, bounds_type_name) in \
+                coords_all_dtypes_and_lazynesses(self, AuxCoord):
+
+            sub_coord = main_coord[:2, 1]
+
+            msg = ('Indexing coord of dtype {} '
+                   'with {} points and {} bounds '
+                   'changed laziness of {} from {!r} to {!r}.')
+            coord_dtype = main_coord.dtype
+            sub_points_lazyness = lazyness_string(sub_coord.core_points())
+            self.assertEqual(
+                sub_points_lazyness, points_type_name,
+                msg.format(coord_dtype,
+                           points_type_name, bounds_type_name,
+                           'points', points_type_name, sub_points_lazyness))
+
+            if bounds_type_name is not 'no':
+                sub_bounds_lazy = lazyness_string(sub_coord.core_bounds())
+                self.assertEqual(
+                    sub_bounds_lazy, bounds_type_name,
+                    msg.format(coord_dtype,
+                               points_type_name, bounds_type_name,
+                               'bounds', bounds_type_name, sub_bounds_lazy))
+
+    def test_real_data_copies(self):
```

Comment:
> don't we want to run a similar test on lazy data?

I don't think that lazy data has any obvious way of checking that a distinct copied array is actually mapped to the same data.
Also, unlike with real data, copying should not create any significant performance problems.  So I think we don't need to check this.